### PR TITLE
New docs

### PR DESCRIPTION
**Pull request by ajaysub110**

### CODE REVIEW

**Comment by ajaysub110** [comment]
on what?

#itsokayboss

**Comment by Sharad24** [comment]
Just GenRLly

**Comment by ajaysub110** [comment]
I can spend time on docs cause it doesn't require continuous periods of thinking time, if that's okay

**Comment by Sharad24** [comment]
Are docs building for you locally?

**Comment by ajaysub110** [comment]
Its building. But new docs are not getting updated. Trying to fix that

File: .travis.yml [hunk 1]
```diff
@@ -47,9 +47,10 @@ jobs:
 
       install:
         - pip install -r docs/requirements.txt
+        - python setup.py install
 
       script:
-        - make -C docs/
+        - make clean -C docs/ html
```
Comment (Sharad24) [inline .travis.yml:53]: Was this the error?

Comment (ajaysub110) [reply .travis.yml:53]: Yeah

**Comment by ajaysub110** [comment]
Package needs to be installed everytime before running `make html`. Now docs get updated with changes.

File: .travis.yml (hunk 1 above)
Comment (Sharad24) [reply .travis.yml:53]: Hmm weird, never done this before

**Comment by Sharad24** [comment]
Feel free to merge, when done

**Comment by ajaysub110** [comment]
@Sharad24 Can you check if this works locally for you now?If it does, let's merge. I'll add readthedocs.io as the webpage and then open another PR to add docs to all the other classes and methods. Does that sound good?

**Comment by Sharad24** [comment]
Yeah.

**Comment by ajaysub110** [comment]
Cool. Merging

**Comment by Sharad24** [comment]
Wait

**Comment by Sharad24** [comment]
Lets wait for travis to finish, I'll test it too

**Comment by ajaysub110** [comment]
Oh I thought you already tested, Sorry

File: .travis.yml (hunk 1 above)
Comment (ajaysub110) [reply .travis.yml:53]: sphinx-quickstart creates a makefile to automatically build the rst files

**Comment by Sharad24** [comment]
Yeah builds for me, but with a lot of warnings. But still documentation looks pretty good to be honest.

**Comment by ajaysub110** [comment]
Yeah, looks pretty good. I think the warnings will go once all the docstrings are "legal"

**Comment by Sharad24** [comment]
We definitely need to structure everything better though. Modules, sub-modules, etc

**Comment by ajaysub110** [comment]
I think, we'll need to set what we want the user to see in the toc tree (table of contents)

**Comment by Sharad24** [comment]
Yup, right

**Comment by ajaysub110** [comment]
Travis done. Shall I merge?

**Comment by Sharad24** [comment]
Yeah!

**Pull request merged**